### DISCUSSION TITLE
serial reaction time plugin - trial ends after specific keypress

### DISCUSSION THREAD

**Discussion by vekteo** [jspsych/jsPsych#3370] [Q&A]
Hi jsPsych community,

I'm working on a serial reaction time task using the jsPsych plugin, and I'm encountering a challenge. I want to ensure that the next trial only begins after a specific button is pressed. By default (if I am using response_ends_trial = true), any keypress seems to trigger the next trial. Also, I would like to record all keypresses until the correct keypress.

Is there a recommended way to achieve this behavior? What I have tried so far is to insert new conditional trials after an incorrect trial, but I am not sure if it is good practice.

**Comment by ChristopheBossens** [comment]
Hi Vekteo,

Do I understand correctly that you only want the trial to end when the correct response has been made? In combination with your need to collect all keypresses, I think it would be the easiest to make some small modifications to the code of the plugin itself . Let me know if you need assistance with this.
